conans/client/file_copier.py

```diff
@@ -51,7 +52,7 @@ def report(self, output):
         return report_copied_files(self._copied, output)
 
     def __call__(self, pattern, dst="", src="", keep_path=True, links=False, symlinks=None,
-                 excludes=None, ignore_case=True):
+                 excludes=None, ignore_case=True, if_different=True):
```

Comment:
It is not clear in which scenarios it will be copying a file and the target already exists. The idea is that the ``self.copy()`` copies to an empty package folder, so this shouldn't be an issue.

Why is this being added? Performance?

In any case, it is extremely recommended to not change the current default, as the likelihood of breaking is much much higher. In this case the default should be ``if_different=False``

Reply:
I made this draft PR to start a discussion, that is why there are no tests so far.
My understanding is that this function is being used in the conan imports method as well.

Use case: 
When doing local development, it is very common to reuse the same build directory. Adding this new param `if_different` will optimize copying files to local build directory, specially when:
* large files needs to be copied to the build directory
* Some files in the build directory are already used by another process (currently the import fail, even there is no need to re import this file as it already exist)

Reply:
Understood, thanks, yes, it makes sense.

Some hints about the future though:

- We will make explicit the ``self.copy()``, so maybe this could be added to the imports case only
- In Conan 2.0 we will probably reduce or integrate the ``imports`` case with the ``deploy`` case. The general idea is that copying files over and over is indeed inefficient, and with virtualenvs it is not needed in most cases, specially development cases. It will only be needed for final extraction of files for final deployment, but not for regular development.